rust/src/mapmanip/core/mod.rs

```diff
@@ -0,0 +1,145 @@
+pub mod to_dict_map;
+pub use to_dict_map::to_dict_map;
+pub mod to_grid_map;
+pub use to_grid_map::to_grid_map;
+
+pub mod map_to_string;
+pub use map_to_string::map_to_string;
+
+use dmmtools::dmm;
+use dmmtools::dmm::Coord3;
+
+///
+#[derive(Clone, Debug, Default)]
+pub struct Tile {
+    ///
+    pub key_suggestion: dmm::Key,
+    ///
+    pub prefabs: Vec<dmm::Prefab>,
+}
+
+impl Tile {
+    ///
+    pub fn get_area(&self) -> Option<&dmm::Prefab> {
+        self.prefabs
+            .iter()
+            .find(|prefab| prefab.path.starts_with("/area/"))
+    }
+
+    ///
+    pub fn remove_area(&mut self) -> Option<dmm::Prefab> {
+        let area = self.get_area().cloned();
+        if area.is_some() {
+            self.prefabs
+                .retain(|prefab| !prefab.path.starts_with("/area/"))
+        }
+        area
+    }
+
+    ///
+    pub fn get_turf(&self) -> Option<&dmm::Prefab> {
+        self.prefabs
+            .iter()
+            .find(|prefab| prefab.path.starts_with("/turf/"))
+    }
+
+    ///
+    pub fn remove_turf(&mut self) -> Option<dmm::Prefab> {
+        let turf = self.get_turf().cloned();
+        if turf.is_some() {
+            self.prefabs
+                .retain(|prefab| !prefab.path.starts_with("/turf/"))
+        }
+        turf
+    }
+}
+
+/// Thin abstraction over a flat vec, to provide a simple hashmap-like interface,
+/// and to translate between 3D dmm coords (start at 1), and 1D flat vec coords (start at 0).
+/// The translation is so that it looks better in logs/errors/etc,
+/// where shown coords would correspond to coords seen in game or in strongdmm.
+#[derive(Clone, Debug)]
+pub struct TileGrid {
+    pub size: Coord3,
+    pub grid: Vec<crate::mapmanip::core::Tile>,
+}
+
+impl TileGrid {
+    pub fn new(size_x: i32, size_y: i32, size_z: i32) -> TileGrid {
+        Self {
+            size: Coord3::new(size_x, size_y, size_z),
+            grid: vec![Tile::default(); (size_x * size_y * size_y) as usize],
```

Comment:
Is this what you meant here?
```suggestion
            grid: vec![Tile::default(); (size_x * size_y * size_z) as usize],
```

Reply:
Yup: https://github.com/Aurorastation/Aurora.3/pull/20029